controllers/discordactions.js

```diff
@@ -66,9 +67,24 @@ const createGroupRole = async (req, res) => {
 const getAllGroupRoles = async (req, res) => {
   try {
     const { groups } = await discordRolesModel.getAllGroupRoles();
+    const groupsWithMemberCount = await discordRolesModel.getNumberOfMemberForGroups(groups);
+    const groupCreatorIds = groupsWithMemberCount.reduce((ids, group) => {
+      ids.add(group.createdBy);
+      return ids;
+    }, new Set());
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
+    const groupsWithUserDetails = groupsWithMemberCount.map((group) => {
+      return {
+        ...group,
+        firstName: groupCreatorsDetails[group.createdBy].first_name,
```

Comment:
since ` groupCreatorsDetails[group.createdBy]` is being used in multiple places then it can be stored in a const.
It would look clean.
```
createdBy = groupCreatorsDetails[group.createdBy]
```

Reply:
Yes. That's a good catch.
I somehow missed it.

Thank you